gradle.properties

```diff
@@ -1,3 +1,5 @@
 org.gradle.jvmargs='-Dfile.encoding=UTF-8'
 android.enableJetifier=true
 android.useAndroidX=true
+org.gradle.configureondemand=true
+org.gradle.parallel=true
```

Comment:
The concurrent task execution which this enables causes problems for artifact publishing to Sonatype which sees the concurrent uploads as separate projects. If someone wants to use this unstable feature, they can enable it locally rather than us forcing it on them.

Reply:
Thanks, I wasn't aware of that. I'll remove this configuration.

Reply:
Separately, if you want to be able to publish in parallel you can use https://github.com/gradle-nexus/publish-plugin/.

Reply:
Thanks for the link. One complication is we need to retain the ability to deploy to our internal Artifactory with a simple env var change ideally.

Reply:
That indeed complicates things. To get the best of both, you'd need to one way or another have different logic. I guess this might be further complicated by the requirement to not have square-private publishing be part of the “open” source repository.

If I were tasked with implementing this, I'd probably use the nexus-publish plugin to publish to OSSRH as it's the only reliable way I have found to avoid ending up with multiple staging repositories, and use the Artifactory plugin to upload to Artifactory. Its primary benefit is that it uses finer grained parallelism for uploading than the stock plugins. 

If you don't want to commit such Square specifics to the repo, the Artifactory functionality would need to be wrapped in a plugin that is only injected in the Square environment one way or another. I imagine such a thing could be used by any other open Square projects that build with Gradle. 

This wouldn't really facilitate arbitrarily changing the destination repo _address_ with an env var, but you could use that to rewire what a `publish` lifecycle task does (i.e. publish to public or private) based on such an external input. Alternatively, you could just have `publishPublic` and `publishPrivate` lifecycle tasks etc. Or, you could have have dual Sonatype/Artifactory capabilities, and allow use of arbitrary destination repos along those lines, controlled via env var.

Or of course, stick with the current lowest-common-denominator approach at the expense of build time.